usso/openid.go

```diff
@@ -78,15 +77,18 @@ func LoginHandler(w http.ResponseWriter, r *http.Request) {
 
 	resp, err := verify(url.String())
 	if err != nil {
+		// A mangled OpenID response is suspicious, so leave a nasty response
 		replyHTTPError(w, http.StatusBadRequest, err)
 		return
 	}
 
-	// get form username and get from datastore the User
+	// Check we have the mandatory fields in the OpenID response
 	username := r.Form.Get("openid.sreg.nickname")
-	if len(username) == 0 {
-		log.Println("Got no 'openid.sreg.nickname' in response params")
-		replyHTTPError(w, http.StatusBadRequest, errgo.New("OpenID response has not valid format"))
+	fullname := r.Form.Get("openid.sreg.fullname")
+	email := r.Form.Get("openid.sreg.email")
+	if len(username) == 0 || len(fullname) == 0 || len(email) == 0 {
```

Comment:
I think (correct me if I'm wrong) that strictly needed can only be the nickname. And in case fullname is empty, we could fill User.Name with nickname content. Email i guess that could be empty if not received